### PR TITLE
Code base cleanup #6065

### LINKED ISSUE #6065: Code base cleanup
There are many mistakes/non-standard practices that have been in the repo since time immemorial, and some that are newly introduced.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #6065 

Most of the stuffs done can be identified via commit messages. Here are some explanations where necessary:
- 2nd commit: the two files `datepicker.css` and `StudentProfileEditSaveActionTest.java` contains `\r` (not `\n`, not `\r\n`) which is not an acceptable line break. In addition, the standardizing-line-break in `HtmlHelper` is unnecessary as the system's line break can be used to do the job.
- 5th commit: no longer relevant.
- 6th commit: either there are workarounds so they are not needed or they are simply unnecessary.
- 7th commit: Eclipse has this "Clean Up..." option under "Source" menu, which has some seriously good functions.

### CODE REVIEW

**Comment by junhaoyap** [comment]
Looking commit at commit

**Comment by junhaoyap** [comment]
Fell asleep, looking commit by commit now :p

**Comment by junhaoyap** [comment]
1. Ok - that's strange indeed, I think they look for specifically ```@AfterClass``` and ignore anything afterwards which is why it's still valid
2. Got it - I was about to ask the same whether it's a change from something to ```\n``` or sorts
3. I'll just assume you used Eclipse to clean it up, GitHub doesn't show me which is tabs and which is space so good to go
4. Looks good
5. OK - but you should right click ```@formatter``` it links to someone's profile ;)
6. Got it
7. Wow, Eclipse cleanup is pretty smart.. I am amazed at some of the changes it did :O
8. Not sure why this is there? Not a problem though, since the tests pass :)

**Comment by junhaoyap** [comment]
Just some stuff we can do here:

I'll comment in commit 7 - and for 8 you can just drop a small note so I know how it happened

File: src/main/java/teammates/common/datatransfer/FeedbackConstantSumResponseDetails.java [hunk 1]
```diff
@@ -18,9 +18,9 @@ public FeedbackConstantSumResponseDetails() {
     public void extractResponseDetails(FeedbackQuestionType questionType,
             FeedbackQuestionDetails questionDetails, String[] answer) {
         List<Integer> constSumAnswer = new ArrayList<Integer>();
-        for (int i = 0; i < answer.length; i++) {
+        for (String element : answer) {
```
Comment (junhaoyap) [inline src/main/java/teammates/common/datatransfer/FeedbackConstantSumResponseDetails.java]: Can we help out the cleaner here and change it to ```for (String answer : answers) {```, more semantical

File: src/main/java/teammates/common/datatransfer/FeedbackRankOptionsResponseDetails.java [hunk 2]
```diff
@@ -24,9 +24,9 @@ public void extractResponseDetails(FeedbackQuestionType questionType,
                                        FeedbackQuestionDetails questionDetails,
                                        String[] answer) {
         List<Integer> rankAnswer = new ArrayList<Integer>();
-        for (int i = 0; i < answer.length; i++) {
+        for (String element : answer) {
```
Comment (junhaoyap) [inline src/main/java/teammates/common/datatransfer/FeedbackRankOptionsResponseDetails.java]: Same here

File: src/main/java/teammates/common/datatransfer/StudentAttributes.java [hunk 3]
```diff
@@ -101,10 +101,8 @@ public StudentAttributes(CourseStudent student) {
         this.lastName = student.getLastName();
         this.comments = Sanitizer.sanitizeTextField(student.getComments());
         this.team = student.getTeamName();
-        this.section = student.getSectionName() == null ? Const.DEFAULT_SECTION
-                                                        : student.getSectionName();
-        this.googleId = student.getGoogleId() == null ? ""
-                                                      : student.getGoogleId();
+        this.section = student.getSectionName() == null ? Const.DEFAULT_SECTION : student.getSectionName();
```
Comment (junhaoyap) [inline src/main/java/teammates/common/datatransfer/StudentAttributes.java]: Can't tell properly via the UI here, does it align as per our code guidelines?

File: src/main/java/teammates/common/util/StringHelper.java [hunk 4]
```diff
@@ -309,8 +309,8 @@ public static String removeExtraSpace(String str) {
     
     private static String byteArrayToHexString(byte[] b) {
         StringBuilder sb = new StringBuilder(b.length * 2);
-        for (int i = 0; i < b.length; i++) {
-            int v = b[i] & 0xff;
+        for (byte element : b) {
```
Comment (junhaoyap) [inline src/main/java/teammates/common/util/StringHelper.java]: This sounds... funny, can we find a better naming?

File: src/main/java/teammates/logic/backdoor/BackDoorLogic.java [hunk 5]
```diff
@@ -479,99 +474,6 @@ private void deleteCourses(Collection<CourseAttributes> courses) {
         }
     }
 
-    //TODO: remove this when we confirm it is not needed
```
Comment (junhaoyap) [inline src/main/java/teammates/logic/backdoor/BackDoorLogic.java]: Did Eclipse remove this, or did you do it?

File: src/main/java/teammates/logic/core/TaskQueuesLogic.java [hunk 6]
```diff
@@ -45,8 +45,8 @@ public void createAndAddTaskMultisetParam(String queueName,
             String name = entry.getKey();
             String[] value = entry.getValue();
             
-            for (int i = 0; i < value.length; i++) {
-                taskToBeAdded = taskToBeAdded.param(name, value[i]);
+            for (String element : value) {
```
Comment (junhaoyap) [inline src/main/java/teammates/logic/core/TaskQueuesLogic.java]: Any better naming?

File: src/main/java/teammates/logic/core/TeamEvalResult.java [hunk 7]
```diff
@@ -347,8 +346,8 @@ private static double averageColumn(double[][] array, int columnIndex) {
         double sum = 0;
         int count = 0;
         StringBuilder values = new StringBuilder();
-        for (int j = 0; j < array.length; j++) {
-            double value = array[j][columnIndex];
+        for (double[] element : array) {
```
Comment (junhaoyap) [inline src/main/java/teammates/logic/core/TeamEvalResult.java]: Doesn't sound very understandable

File: src/main/java/teammates/storage/api/CommentsDb.java [hunk 8]
```diff
@@ -356,20 +356,6 @@ private void updateInstructorEmailAsLastEditorForStudentComments(String courseId
         getPm().close();
     }
     
-    // for now, this method is not being used as instructor cannot be receiver
```
Comment (junhaoyap) [inline src/main/java/teammates/storage/api/CommentsDb.java]: Same, Eclipse or you?

File: src/main/java/teammates/ui/controller/AdminActivityLogPageData.java [hunk 9]
```diff
@@ -127,8 +127,8 @@ public void setToDate(long endTime) {
      * value is converted to lower case before comparing
      */
     private boolean arrayContains(String[] arr, String value) {
-        for (int i = 0; i < arr.length; i++) {
-            if (arr[i].equals(value.toLowerCase().trim())) {
+        for (String element : arr) {
```
Comment (junhaoyap) [inline src/main/java/teammates/ui/controller/AdminActivityLogPageData.java]: Better naming please!

File: src/test/java/teammates/test/cases/ui/browsertests/TableSortTest.java [hunk 10]
```diff
@@ -158,8 +158,8 @@ public void testStableSort() {
         };
 
         StringBuilder searchString = new StringBuilder();
-        for (int i = 0; i < idList.length; i++) {
-            searchString.append(idList[i]).append("{*}");
+        for (String element : idList) {
```
Comment (junhaoyap) [inline src/test/java/teammates/test/cases/ui/browsertests/TableSortTest.java]: Naming

File: src/test/java/teammates/test/cases/ui/browsertests/TableSortTest.java [hunk 11]
```diff
@@ -183,8 +183,8 @@ public void testStableSort() {
         };
 
         searchString = new StringBuilder();
-        for (int i = 0; i < reversedIdList.length; i++) {
-            searchString.append(reversedIdList[i]).append("{*}");
+        for (String element : reversedIdList) {
```
Comment (junhaoyap) [inline src/test/java/teammates/test/cases/ui/browsertests/TableSortTest.java]: Ditto

**Comment by junhaoyap** [comment]
Mostly because the helper doesn't give proper naming, ping me again and I can look at it tomorrow when you're done 👍

File: src/main/java/teammates/logic/backdoor/BackDoorLogic.java (hunk 5 above)
Comment (wkurniawan07) [inline src/main/java/teammates/logic/backdoor/BackDoorLogic.java]: Eclipse did this. Apparently they don't care about the `@SuppressWarnings("unused")`.

File: src/main/java/teammates/storage/api/CommentsDb.java (hunk 8 above)
Comment (wkurniawan07) [inline src/main/java/teammates/storage/api/CommentsDb.java]: Same here, also Eclipse.

File: src/main/java/teammates/common/datatransfer/StudentAttributes.java (hunk 3 above)
Comment (wkurniawan07) [inline src/main/java/teammates/common/datatransfer/StudentAttributes.java]: It's one line so there's nothing to align. Also, this is not Eclipse but my doing, just happened to spot it :p

**Comment by junhaoyap** [comment]
Great - no other problems - let's get it in!

**Pull request merged**